src/pyedb/dotnet/edb_core/layout.py

```diff
@@ -124,11 +147,13 @@ def primitives(self):
         list of :class:`pyedb.dotnet.edb_core.edb_data.primitives_data.EDBPrimitives`
             List of primitives.
         """
-        _prims = []
+        if len(self._primitives) == len(self._layout.primitives):
+            return self._primitives
```

Comment:
Is it possible that we end up adding an element to `self._layout.primitives` and the removing another one, making `self._layout.primitives` having the same length but with a different content ? If yes, then I think we should leverage this if condition as one might end up with incorrect cached data.

Reply:
I've added a cleanup of the variables on the delete operation so we should be safe enough.